staging/src/k8s.io/client-go/util/flowcontrol/backoff.go

```diff
@@ -67,15 +68,45 @@ func (p *Backoff) Get(id string) time.Duration {
 	return delay
 }
 
+// GetWithNbRetry the current backoff Duration and the number of previous retry
+func (p *Backoff) GetWithNbRetry(id string) (time.Duration, int32) {
+	p.Lock()
+	defer p.Unlock()
+	var delay time.Duration
+	var nbRetry int32
+	entry, ok := p.perItemBackoff[id]
+	if ok {
+		delay = entry.backoff
+		nbRetry = entry.retry
+	}
+	return delay, nbRetry
+}
+
 // move backoff to the next mark, capping at maxDuration
 func (p *Backoff) Next(id string, eventTime time.Time) {
 	p.Lock()
 	defer p.Unlock()
 	entry, ok := p.perItemBackoff[id]
 	if !ok || hasExpired(eventTime, entry.lastUpdate, p.maxDuration) {
-		entry = p.initEntryUnsafe(id)
+		entry = p.initEntryUnsafe(id, p.defaultDuration)
+	} else {
+		delay := entry.backoff * 2 // exponential
+		entry.retry++
+		entry.backoff = time.Duration(integer.Int64Min(int64(delay), int64(p.maxDuration)))
+	}
+	entry.lastUpdate = p.Clock.Now()
+}
+
+// NextWithInitDuration move backoff to the next mark and increment the number of retry, capping at maxDuration
+func (p *Backoff) NextWithInitDuration(id string, backoffDuration time.Duration, eventTime time.Time) {
```

Comment:
s/backoffDuration/initial

Let the name suggest what it's doing. Similarly to `NewBackOff`.

Reply:
Again, do not copy&paste code. Create a helper function and call that from `Next` and `NextWithInitDuration` passing `p.defaultDuration` or `initial` accordingly.